gson/src/main/java/com/google/gson/internal/bind/JsonTreeWriter.java

```diff
@@ -199,6 +200,16 @@ private void put(JsonElement value) {
     return this;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @since 2.8.7
+   */
+  @Override public JsonWriter jsonValue(String value) throws IOException {
+    put(JsonParser.parseString(value));
```

Comment:
As pointed out in the previous review comment, could you please check for `null` and then perform`return nullValue();` instead; the documentation demands this behavior, currently it would throw a `NullPointerException`.